### ISSUE TITLE
Does the model have to be trained on all classes in the entire dataset?

### ISSUE THREAD

**Issue by boxbox2** [HuiZhang0812/DiffusionAD#48]
Because training on the entire MVTEC dataset is too time-consuming, I only trained the 'carpet' category.
This is my args.json, batchsize = 4,epochs = 1500，Different from the author's batchsize = 16 and epoch = 3000
```
{
  "img_size": [256,256],
  "Batch_Size": 4,
  "EPOCHS": 1500,
  "T": 1000,
  "base_channels": 128,
  "beta_schedule": "linear",
  "loss_type": "l2",
  "diffusion_lr": 1e-4,
  "seg_lr": 1e-5,
  "random_slice": true,
  "weight_decay": 0.0,
  "save_imgs":true,
  "save_vids":false, 
  "dropout":0,
  "attention_resolutions":"32,16,8",
  "num_heads":4,
  "num_head_channels":-1,
  "noise_fn":"gauss",
  "channels":3,
  "mvtec_root_path":"/workspace/DiffusionAD/MVTec-AD",
  "visa_root_path":"/workspace/DiffusionAD/VisA",
  "dagm_root_path":"/workspace/DiffusionAD/dagm",
  "mpdd_root_path":"/workspace/DiffusionAD/mpdd",
  "anomaly_source_path":"/workspace/DiffusionAD/dtd",
  "noisier_t_range":600,
  "less_t_range":300,
  "condition_w":1,
  "eval_normal_t":200,
  "eval_noisier_t":400,
  "output_path":"/workspace/DiffusionAD/outputs"

}

```
after 1500,train loss is 3.232
![trans_loss](https://github.com/HuiZhang0812/DiffusionAD/assets/56194614/bcba00c1-2b1e-404b-b8c9-04af72dc911f)

**when i run eval.py,it will say missing key**
![image](https://github.com/HuiZhang0812/DiffusionAD/assets/56194614/6246a4fb-acb8-44d2-953b-806c7dfb6b2b)
so i changed the eval.py,(line 381)
```
        unet_model.load_state_dict(output["unet_model_state_dict"],strict=False)
        unet_model.to(device)
```
after this
![image](https://github.com/HuiZhang0812/DiffusionAD/assets/56194614/3ea217e9-46ae-4bac-8980-ad150ee3110b)
**It can be found that out_mask has better results, but recon_con is still a Gaussian image.**

**Isn’t this training a model per category? Why can Segmentation Sub-network training be successful, but Norm-guided One-step Denoising does not have good results?Or is it because I didn’t train enough rounds? Or must all categories be trained to avoid possible missing keys problems?**                    

Looking forward to your reply

**Comment by boxbox2** [comment]
One more thing, because my single card only 16g  so I changed to multiple card to run unet_model，but the seg_model didn't.
![image](https://github.com/HuiZhang0812/DiffusionAD/assets/56194614/c07caccb-050e-4fcf-a8b7-1cf6ced5e689)
Maybe it doesn't support multi-card running?I don't know. Has anyone encountered and solved this problem？

**Comment by boxbox2** [comment]
I solved it.
![image](https://github.com/HuiZhang0812/DiffusionAD/assets/56194614/a72e9f15-62b8-4c64-89b5-324ae7dcc8ea)
in the past，When I selected multi-card when training, but eval.py did not write multi-card function, the problem of missing key will occur. I chose to exercise strict = false at 385. But something went wrong. I solved the problem by adding 381 lines in eval.py
![image](https://github.com/HuiZhang0812/DiffusionAD/assets/56194614/3af1014d-497d-4c7a-9d65-d05a19e73580)

**Comment by wasdhcb** [comment]
> only trained the 'carpet' category

how to train a single mvtec category?
